File: CallX2/app/src/main/java/com/callx/app/adapters/MediaThumbAdapter.java
package com.callx.app.adapters;

import android.view.*;
import android.widget.*;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.callx.app.R;
import java.util.List;

/**
 * MediaThumbAdapter — 3-column media grid for GroupInfoActivity.
 * Each cell shows a square thumbnail loaded via Glide.
 */
public class MediaThumbAdapter extends RecyclerView.Adapter<MediaThumbAdapter.VH> {

    public interface OnMediaClickListener { void onClick(String url); }

    private final List<String>        urls;
    private final OnMediaClickListener listener;

    public MediaThumbAdapter(List<String> urls, OnMediaClickListener listener) {
        this.urls     = urls;
        this.listener = listener;
    }

    @NonNull @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // Square cell — dynamically sized to 1/3 of screen width
        int size = parent.getMeasuredWidth() / 3;
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_media_thumb, parent, false);
        RecyclerView.LayoutParams lp = new RecyclerView.LayoutParams(size, size);
        v.setLayoutParams(lp);
        return new VH(v);
    }

    @Override
    public void onBindViewHolder(@NonNull VH h, int pos) {
        String url = urls.get(pos);
        Glide.with(h.ivThumb.getContext())
                .load(url)
                .centerCrop()
                .placeholder(R.drawable.ic_gallery)
                .into(h.ivThumb);
        h.itemView.setOnClickListener(v -> listener.onClick(url));
    }

    @Override public int getItemCount() {
        return Math.min(urls.size(), 9); // Show max 9 in info screen
    }

    static class VH extends RecyclerView.ViewHolder {
        ImageView ivThumb;
        VH(@NonNull View itemView) {
            super(itemView);
            ivThumb = itemView.findViewById(R.id.iv_media_thumb);
        }
    }
}

File: CallX2/feature-chat/src/main/java/com/callx/app/adapters/MediaThumbAdapter.kt
package com.callx.app.adapters

import android.view.*
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.callx.app.chat.R

/**
 * MediaThumbAdapter — 3-column media grid for GroupInfoActivity.
 */
class MediaThumbAdapter(
    private val urls: List<String>,
    private val listener: OnMediaClickListener
) : RecyclerView.Adapter<MediaThumbAdapter.VH>() {

    interface OnMediaClickListener { fun onClick(url: String) }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val size = parent.measuredWidth / 3
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_media_thumb, parent, false)
        v.layoutParams = RecyclerView.LayoutParams(size, size)
        return VH(v)
    }

    override fun onBindViewHolder(h: VH, pos: Int) {
        val url = urls[pos]
        Glide.with(h.ivThumb.context).load(url).centerCrop()
            .placeholder(R.drawable.ic_gallery).into(h.ivThumb)
        h.itemView.setOnClickListener { listener.onClick(url) }
    }

    override fun getItemCount() = minOf(urls.size, 9)

    class VH(v: View) : RecyclerView.ViewHolder(v) {
        val ivThumb: ImageView = v.findViewById(R.id.iv_media_thumb)
    }
}
